src/theory/bags/bags_rewriter.cpp

```diff
@@ -648,6 +650,18 @@ BagsRewriteResponse BagsRewriter::postRewriteFold(const TNode& n) const
   return BagsRewriteResponse(n, Rewrite::NONE);
 }
 
+BagsRewriteResponse BagsRewriter::postRewritePartition(const TNode& n) const
+{
+  Assert(n.getKind() == kind::BAG_PARTITION);
+  if (n[1].isConst())
+  {
+    Node ret = BagsUtils::evaluateBagPartition(d_rewriter, n);
+    return BagsRewriteResponse(ret, Rewrite::PARTITION_CONST);
```

Comment:
Can you guard this `ret!=n` to make the (non-)use of the rewrite accurate?

Reply:
Right. Done.